fenix/app/src/main/java/org/mozilla/fenix/home/TabsTrayReducer.kt

```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home
```

Comment:
Maybe these reducers should be in the same package as the appstate (maybe something like appstate/home)?

Reply:
I am happy to do this in the interim if you would like, but my thoughts for the long-term are that we would want state, reducers, and actions to live in individual feature modules and then be composed into the single store.  So this was intended to mirror that future, rather than group everything related to State into a single package. One issue here is that we already have a `TabsTrayReducer` for the existing `TabsTrayStore`, so we need some kind of distinction between the two for now.

What the future might look::

```
// in Tabs Tray module
data class TabsTrayState(...)

sealed class TabsTrayAction {
    object NewTab : TabsTrayAction
}

object TabsTrayReducer {
    fun reduce(...)
}

// in top level module
data class AppState(
    val tabsTrayState: TabsTrayState
)

// need a dedicated wrapping action so that we can still have a sealed class and compile-time exhaustiveness checking
sealed class AppAction {
    data class TabsTrayAppAction(val inner: TabsTrayAction)
}

object AppReducer {
    fun reduce(state: AppState, action: AppAction): AppState = when (action) {
        is TabsTrayAppAction -> TabsTrayReducer.reduce(action.inner)
    }
}
```

Reply:
**Aside from the actual PR:** I am curious if this is something that you tried prototyping to validate that it will work. I must've tried to do something similar with these Stores many years ago right before pocmo left along with Christian to see if we could have a single Store, but couldn't get it to work. I don't really remember what it is anymore, but the general idea I am trying to convey here is to make sure said high-level architecture could work.

Reply:
I did some prototyping earlier last year, but unfortunately lost it when my workstation was destroyed. I do firmly believe it's possible and is just a matter of improving our existing infrastructure. It's a high background priority for me, and would definitely expect a robust prototype and RFC stage before we made any commitments to move forward with something like that.